app/src/main/java/org/wikipedia/dataclient/mwapi/MwQueryResult.kt

```diff
@@ -0,0 +1,171 @@
+package org.wikipedia.dataclient.mwapi
+
+import com.google.gson.annotations.SerializedName
+import org.apache.commons.lang3.StringUtils
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.json.PostProcessingTypeAdapter.PostProcessable
+import org.wikipedia.notifications.Notification
+import org.wikipedia.notifications.Notification.SeenTime
+import org.wikipedia.notifications.Notification.UnreadNotificationWikiItem
+import org.wikipedia.page.PageTitle
+import org.wikipedia.settings.SiteInfo
+import org.wikipedia.util.DateUtil
+import java.util.*
+
+class MwQueryResult : PostProcessable {
+
+    @SerializedName("userinfo") val userInfo: UserInfo? = null
+    @SerializedName("unreadnotificationpages") val unreadNotificationWikis: Map<String, UnreadNotificationWikiItem>? = null
+    @SerializedName("authmanagerinfo") private val amInfo: MwAuthManagerInfo? = null
+    @SerializedName("general") val siteInfo: SiteInfo? = null
+    @SerializedName("wikimediaeditortaskscounts") val editorTaskCounts: EditorTaskCounts? = null
+    @SerializedName("usercontribs") val userContributions: List<UserContribution> = emptyList()
+
+    private val redirects: MutableList<Redirect>? = null
+    private val converted: MutableList<ConvertedTitle>? = null
+    private val users: List<ListUserResponse>? = null
+    private val tokens: Tokens? = null
+    private val echomarkread: MarkReadResponse? = null
+    val pages: MutableList<MwQueryPage>? = null
+    val echoMarkSeen: MarkReadResponse? = null
+    val notifications: NotificationList? = null
+    val watchlist: List<WatchlistItem> = emptyList()
+
+    fun firstPage(): MwQueryPage? {
+        return if (pages != null && pages.size > 0) {
+            pages[0]
+        } else null
+    }
+
+    fun csrfToken(): String? {
+        return tokens?.csrf
+    }
+
+    fun watchToken(): String? {
+        return tokens?.watch
+    }
+
+    fun createAccountToken(): String? {
+        return tokens?.createAccount
+    }
+
+    fun loginToken(): String? {
+        return tokens?.login
+    }
+
+    fun captchaId(): String? {
+        var captchaId: String? = null
+        amInfo?.requests?.asSequence()?.filter { "CaptchaAuthenticationRequest" == it.id }?.forEach { captchaId = it.fields!!["captchaId"]!!.value }
+        return captchaId
```

Comment:
Can it be simplified to `amInfo?.requests?.find { "CaptchaAuthenticationRequest" == it.id }?.fields?.get("captchaId")?.value`?